pkg/systemd/host.js

```diff
@@ -508,6 +508,16 @@ PageServer.prototype = {
             var os_updates = { };
             self.os_updates = null;
 
+            // skip update check when becoming invisible
+            if (document.visibilityState != "visible")
+                return;
+            const lastCheck = window.sessionStorage.getItem("last_sw_update_check");
+            // check at most once per hour, as this is expensive on embedded machines
+            // this gets reset on applying software updates updates
+            if (lastCheck && Date.now() - lastCheck < 3600000)
+                return;
```

Comment:
This also skips the check on page reload, and then the line stays empty.  I think checking `self.os_updates` for `null` should work.

Reply:
```diff
diff --git a/pkg/systemd/host.js b/pkg/systemd/host.js
index b0ac47b46..3ccb0400e 100644
--- a/pkg/systemd/host.js
+++ b/pkg/systemd/host.js
@@ -506,7 +506,6 @@ PageServer.prototype = {
 
         function check_for_updates() {
             var os_updates = { };
-            self.os_updates = null;
 
             // skip update check when becoming invisible
             if (document.visibilityState != "visible")
@@ -514,10 +513,11 @@ PageServer.prototype = {
             const lastCheck = window.sessionStorage.getItem("last_sw_update_check");
             // check at most once per hour, as this is expensive on embedded machines
             // this gets reset on applying software updates updates
-            if (lastCheck && Date.now() - lastCheck < 3600000)
+            if (self.os_updates && lastCheck && Date.now() - lastCheck < 3600000)
                 return;
             window.sessionStorage.setItem("last_sw_update_check", Date.now());
 
+            self.os_updates = null;
             packagekit.cancellableTransaction(
                 "GetUpdates", [0],
                 function(data) {
```

Reply:
Nice catch, thanks! Fixed.